metaflow/plugins/airflow/airflow.py

```diff
@@ -0,0 +1,670 @@
+        max_error_retries = 0
+        # Different decorators may have different retrying strategies, so take
+        # the max of them.
+        for deco in node.decorators:
+            user_code_retries, error_retries = deco.step_task_retry_count()
+            max_user_code_retries = max(max_user_code_retries, user_code_retries)
+            max_error_retries = max(max_error_retries, error_retries)
+
+        return max_user_code_retries, max_user_code_retries + max_error_retries
+
+    def _get_retry_delay(self, node):
+        retry_decos = [deco for deco in node.decorators if deco.name == "retry"]
+        if len(retry_decos) > 0:
+            retry_mins = retry_decos[0].attributes["minutes_between_retries"]
+            return timedelta(minutes=int(retry_mins))
+        return None
+
+    def _process_parameters(self):
+        parameters = []
+        seen = set()
+        airflow_params = []
+        type_transform_dict = {
+            int.__name__: "integer",
+            str.__name__: "string",
+            bool.__name__: "string",
+            float.__name__: "number",
+            JSONTypeClass.name: "string",
+        }
+        type_parser = {bool.__name__: lambda v: str(v)}
+
+        for var, param in self.flow._get_parameters():
+            # Throw an exception if the parameter is specified twice.
+            norm = param.name.lower()
+            if norm in seen:
+                raise MetaflowException(
+                    "Parameter *%s* is specified twice. "
+                    "Note that parameter names are "
+                    "case-insensitive." % param.name
+                )
+            seen.add(norm)
+
+            # Airflow requires defaults set for parameters.
+            if "default" not in param.kwargs:
+                raise MetaflowException(
+                    "The parameter *%s* does not have a "
+                    "default set. "
+                    "A default is required for parameters when deploying on Airflow."
+                )
+            value = deploy_time_eval(param.kwargs.get("default"))
+            parameters.append(dict(name=param.name, value=value))
+
+            # Setting airflow related param args.
+            param_type = param.kwargs.get("type", None)
+            airflow_param = dict(
+                name=param.name,
+            )
+            param_help = param.kwargs.get("help", None)
+            if value is not None:
+                airflow_param["default"] = value
+            if param_help:
+                airflow_param["description"] = param_help
+            if param_type is not None:
+                if isinstance(param_type, JSONTypeClass):
+                    airflow_param["type"] = type_transform_dict[JSONTypeClass.name]
+                elif param_type.__name__ in type_transform_dict:
+                    airflow_param["type"] = type_transform_dict[param_type.__name__]
+                    if param_type.__name__ in type_parser and value is not None:
```

Comment:
Does this line need to be nested within the `elif` block?

Reply:
Refactored here for simplicity.